File: api/src/main/java/kor/toxicity/questadder/api/gui/GuiExecutor.java
package kor.toxicity.questadder.api.gui;

import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public interface GuiExecutor {
    void initialize(@NotNull GuiData data);
    void click(@NotNull GuiData data, @NotNull ItemStack clickedItem, int clickedSlot, boolean isPlayerInventory, @NotNull MouseButton button);
    void end(@NotNull GuiData data);
}

File: plugin/src/main/java/kor/toxicity/questadder/util/gui/GuiExecutor.kt
package kor.toxicity.questadder.util.gui

import org.bukkit.inventory.ItemStack

interface GuiExecutor {
    fun initialize(data: GuiData)
    fun click(data: GuiData, clickedItem: ItemStack, clickedSlot: Int, isPlayerInventory: Boolean, button: MouseButton)
    fun end(data: GuiData)
}